ironfish-cli/src/commands/chain/download.ts

```diff
@@ -134,116 +134,142 @@ export default class Download extends IronfishCommand {
-        format:
-          'Downloading snapshot: [{bar}] {percentage}% | {downloadedSize} / {fileSize} | {speed}/s | ETA: {estimate}',
-      }) as ProgressBar
-
-      bar.start(manifest.file_size, 0, {
-        fileSize,
-        downloadedSize: '0',
-        speed: '0',
-        estimate: TimeUtils.renderEstimate(0, 0, 0),
-      })
-
-      const speed = new Meter()
-      speed.start()
-
       let downloaded = 0
+      try {
+        const statResult = await fsAsync.stat(snapshotPath)
+        if (statResult.isFile()) {
+          downloaded = statResult.size
+        }
+      } catch {
+        downloaded = 0
+      }
 
-      const hasher = crypto.createHash('sha256')
-      const writer = fs.createWriteStream(snapshotPath, { flags: 'w' })
+      if (downloaded < manifest.file_size) {
+        this.log(`Downloading snapshot from ${snapshotUrl} to ${snapshotPath}`)
+
+        const bar = CliUx.ux.progress({
+          barCompleteChar: '\u2588',
+          barIncompleteChar: '\u2591',
+          format:
+            'Downloading snapshot: [{bar}] {percentage}% | {downloadedSize} / {fileSize} | {speed}/s | ETA: {estimate}',
+        }) as ProgressBar
+
+        bar.start(manifest.file_size, 0, {
+          fileSize,
+          downloadedSize: FileUtils.formatFileSize(downloaded),
+          speed: '0',
+          estimate: TimeUtils.renderEstimate(0, 0, 0),
+        })
 
-      const idleTimeout = 30000
-      let idleLastChunk = Date.now()
-      const idleCancelSource = axios.CancelToken.source()
+        const speed = new Meter()
+        speed.start()
 
-      const idleInterval = setInterval(() => {
-        const timeSinceLastChunk = Date.now() - idleLastChunk
+        const idleTimeout = 30000
+        let idleLastChunk = Date.now()
+        const idleCancelSource = axios.CancelToken.source()
 
-        if (timeSinceLastChunk > idleTimeout) {
-          clearInterval(idleInterval)
+        const idleInterval = setInterval(() => {
+          const timeSinceLastChunk = Date.now() - idleLastChunk
 
-          idleCancelSource.cancel(
-            `Download timed out after ${TimeUtils.renderSpan(timeSinceLastChunk)}`,
-          )
-        }
-      }, idleTimeout)
+          if (timeSinceLastChunk > idleTimeout) {
+            clearInterval(idleInterval)
 
-      const response: { data: IncomingMessage } = await axios({
-        method: 'GET',
-        responseType: 'stream',
-        url: snapshotUrl,
-        cancelToken: idleCancelSource.token,
-      })
+            idleCancelSource.cancel(
+              `Download timed out after ${TimeUtils.renderSpan(timeSinceLastChunk)}`,
+            )
+          }
+        }, idleTimeout)
+
+        const response: { data: IncomingMessage } = await axios({
+          method: 'GET',
+          responseType: 'stream',
+          url: snapshotUrl,
+          cancelToken: idleCancelSource.token,
+          headers: {
+            range: `bytes=${downloaded}-${manifest.file_size - 1}`,
```

Comment:
Nit: you could use ``range: `bytes=${downloaded}-` `` (i.e. omit the range end) to avoid calculations